File: js/recuperar_contraseña.js
document.querySelector("form").addEventListener("submit", async (e) => {
  e.preventDefault();

  const email = document.getElementById("email").value;

  if (!email) {
    alert("Por favor ingresa un correo válido");
    return;
  }

  try {
    const res = await fetch(" ", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ email }),
    });

    const data = await res.json();

    if (res.ok) {
      alert("✅ Revisa tu correo, hemos enviado el enlace para restablecer tu contraseña");
    } else {
      alert("⚠️ Error: " + data.message);
    }
  } catch (error) {
    console.error("Error al enviar:", error);
    alert("❌ No se pudo enviar el correo, intenta de nuevo.");
  }
});

File: python/recuperar_contraseña.py
from flask import Flask, request, jsonify
from flask_cors import CORS 
from email.message import EmailMessage
import smtplib

app = Flask(__name__)
CORS(app)

@app.route("/api/recuperar", methods=["POST"])
def recuperar():
    try:
        data = request.get_json()
        email_usuario = data.get("email")

        if not email_usuario:
            return jsonify({"message": "Correo no válido"}), 400
        remitente = "soporteagricord@gmail.com"
        clave = "zvsa huun ewyn rwuf"
        mensaje = f"Hola, {email_usuario}. Haz clic en el enlace para recuperar tu contraseña."

        # Configuración del correo
        email = EmailMessage()
        email["From"] = remitente
        email["To"] = email_usuario
        email["Subject"] = "Recuperación de contraseña"
        email.set_content(mensaje)

        smtp = smtplib.SMTP_SSL("smtp.gmail.com")
        smtp.login(remitente, clave)
        smtp.sendmail(remitente, email_usuario, email.as_string())
        smtp.quit()

        return jsonify({"message": "Correo enviado correctamente"}), 200

    except Exception as e:
        print("Error en el Backend (Flask):", e)
        return jsonify({"message": "Error al enviar correo. Revisa la consola del servidor."}), 500

if __name__ == "__main__":
    app.run(port=3000, debug=True)